src/App.tsx

```diff
@@ -26,70 +26,111 @@ type ReorderOptions = {
+  if (sortType === SortType.ALPHABET) {
+    visibleGoods.sort((good1, good2) => good1.localeCompare(good2));
+  }
+
+  if (sortType === SortType.LENGTH) {
+    visibleGoods.sort((a, b) => {
+      return a.length - b.length;
+    });
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
 
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
+export class App extends Component<{}, State> {
+  state = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
 
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
+  render() {
+    const goodsToRender = getReorderedGoods(goodsFromServer, this.state);
 
-        <button
-          type="button"
-          className="button is-danger is-light"
+    return (
+      <div className="section content">
+        <div
+          className="buttons"
+          role="directory"
         >
-          Reset
-        </button>
-      </div>
+          <button
+            type="button"
+            className={this.state.sortType === SortType.ALPHABET
+              ? 'button is-info' : 'button is-info is-light'}
+            onClick={() => {
+              this.setState({ sortType: SortType.ALPHABET, isReversed: false });
+            }}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className={this.state.sortType === SortType.LENGTH
+              ? 'button is-success' : 'button is-success is-light'}
+            onClick={() => {
+              this.setState({ sortType: SortType.LENGTH, isReversed: false });
+            }}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={this.state.isReversed
+              ? 'button is-warning' : 'button is-warning is-light'}
+            onClick={() => {
+              this.setState((state) => ({ isReversed: !state.isReversed }));
+            }}
+          >
+            Reverse
+          </button>
+
+          {(this.state.sortType !== SortType.NONE || this.state.isReversed)
+          && (
+            <button
+              type="button"
+              className="button is-danger is-light"
+              onClick={() => {
+                this.setState({ isReversed: false, sortType: SortType.NONE });
+              }}
```

Comment:
It's better to create separate functions and use them for `onClick`.

Reply:
not fixed
https://github.com/mate-academy/react_list-of-goods/pull/1694/files#r1179384047